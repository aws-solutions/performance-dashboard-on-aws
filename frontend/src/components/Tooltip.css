.padding-x-2 {
  padding-left: 16px !important;
  padding-right: 16px !important;
}
