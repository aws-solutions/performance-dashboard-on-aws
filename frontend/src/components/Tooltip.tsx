/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import ReactTooltip, { Place, Type, Effect, Offset, GetContent } from "react-tooltip";

interface Props {
    id: string;
    uuid?: string;
    place?: Place;
    type?: Type;
    effect?: Effect;
    offset?: Offset;
    getContent?: GetContent;
    clickable?: boolean;
}

function Tooltip(props: Props) {
    return (
        <ReactTooltip
            id={props.id}
            uuid={props.uuid}
            place={props.place}
            type={props.type}
            effect={props.effect}
            offset={props.offset}
            getContent={props.getContent}
            clickable={props.clickable}
            globalEventOff="click"
            className="padding-x-2-important cursor-default shadow-3"
            borderColor="#dfe1e2"
            role="tooltip"
        />
    );
}

export default Tooltip;
