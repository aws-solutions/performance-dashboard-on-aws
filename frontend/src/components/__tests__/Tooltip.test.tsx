/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import Tooltip from "../Tooltip";

test("renders a tooltip at the top (default)", async () => {
    const wrapper = render(<Tooltip id="test" uuid="test" />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a tooltip at the right", async () => {
    const wrapper = render(<Tooltip id="test" uuid="test" place="right" />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a tooltip at the bottom", async () => {
    const wrapper = render(<Tooltip id="test" uuid="test" place="bottom" />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders a tooltip at the left", async () => {
    const wrapper = render(<Tooltip id="test" uuid="test" place="left" />);
    expect(wrapper.container).toMatchSnapshot();
});
