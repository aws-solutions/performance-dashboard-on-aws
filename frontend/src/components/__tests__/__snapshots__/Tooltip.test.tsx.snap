// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a tooltip at the bottom 1`] = `
<div>
  <div
    class="__react_component_tooltip test place-bottom type-dark"
    data-id="tooltip"
    id="test"
    role="tooltip"
  >
    <style
      aria-hidden="true"
    >
      
  	.test {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.test.place-top {
        margin-top: -10px;
    }
    .test.place-top::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .test.place-top::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .test.place-bottom {
        margin-top: 10px;
    }
    .test.place-bottom::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .test.place-bottom::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .test.place-left {
        margin-left: -10px;
    }
    .test.place-left::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-left::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .test.place-right {
        margin-left: 10px;
    }
    .test.place-right::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-right::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  
    </style>
  </div>
</div>
`;

exports[`renders a tooltip at the left 1`] = `
<div>
  <div
    class="__react_component_tooltip test place-left type-dark"
    data-id="tooltip"
    id="test"
    role="tooltip"
  >
    <style
      aria-hidden="true"
    >
      
  	.test {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.test.place-top {
        margin-top: -10px;
    }
    .test.place-top::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .test.place-top::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .test.place-bottom {
        margin-top: 10px;
    }
    .test.place-bottom::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .test.place-bottom::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .test.place-left {
        margin-left: -10px;
    }
    .test.place-left::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-left::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .test.place-right {
        margin-left: 10px;
    }
    .test.place-right::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-right::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  
    </style>
  </div>
</div>
`;

exports[`renders a tooltip at the right 1`] = `
<div>
  <div
    class="__react_component_tooltip test place-right type-dark"
    data-id="tooltip"
    id="test"
    role="tooltip"
  >
    <style
      aria-hidden="true"
    >
      
  	.test {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.test.place-top {
        margin-top: -10px;
    }
    .test.place-top::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .test.place-top::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .test.place-bottom {
        margin-top: 10px;
    }
    .test.place-bottom::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .test.place-bottom::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .test.place-left {
        margin-left: -10px;
    }
    .test.place-left::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-left::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .test.place-right {
        margin-left: 10px;
    }
    .test.place-right::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-right::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  
    </style>
  </div>
</div>
`;

exports[`renders a tooltip at the top (default) 1`] = `
<div>
  <div
    class="__react_component_tooltip test place-top type-dark"
    data-id="tooltip"
    id="test"
    role="tooltip"
  >
    <style
      aria-hidden="true"
    >
      
  	.test {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.test.place-top {
        margin-top: -10px;
    }
    .test.place-top::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .test.place-top::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .test.place-bottom {
        margin-top: 10px;
    }
    .test.place-bottom::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .test.place-bottom::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .test.place-left {
        margin-left: -10px;
    }
    .test.place-left::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-left::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .test.place-right {
        margin-left: 10px;
    }
    .test.place-right::before {
        content: "";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .test.place-right::after {
        content: "";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  
    </style>
  </div>
</div>
`;
